<launch>
    <arg name="node_type"/>
    <arg name="node_id"/>
    <!-- Load the controller -->
    <node name="lee_position_controller_node" pkg="rotors_control" type="lee_position_controller_node" output="screen">
        <rosparam command="load" file="$(find rotors_gazebo)/resource/lee_controller_firefly.yaml" />
        <rosparam command="load" file="$(find rotors_gazebo)/resource/firefly.yaml" />
        <remap from="odometry" to="odometry_sensor1/odometry" />
    </node>
    <!-- Load the navigator -->
    <node name="navigator" pkg="multirobot_sim" type="navigator.py" output="screen">
        <param name="node_id" value="$(arg node_id)"/>
        <param name="odom_topic" value="/$(arg node_id)/odometry_sensor1/odometry"/>
        <param name="goal_topic" value="goal"/>
        <param name="path_topic" value="path"/>
    </node>
    <!-- Repeater node-->
    <node name="repeater" pkg="multirobot_sim" type="repeater.py" output="screen">
        <param name="model_name" value="$(arg node_id)"/>
        <param name="source_topic" value="goal"/>
        <param name="target_topic" value="command/pose"/>
        <param name="path_topic" value="path"/>
    </node>
    <!-- Task allocation-->
    <node name="task_allocation" pkg="multirobot_sim" type="task_allocation.py" output="screen">
        <param name="node_id" value="$(arg node_id)"/>
        <param name="node_type" value="$(arg node_type)"/>
        <param name="odom_topic" value="/$(arg node_id)/odometry_sensor1/odometry"/>
    </node>
</launch>